test/jdk/java/util/zip/ZipFile/DeleteTempJar.java

```diff
@@ -72,16 +72,21 @@ public void handle(HttpExchange e) {
                     }
                 }
             });
-        server.start();
 
-        URL url = new URL("jar:http://localhost:"
+        try {
+            server.start();
+            URL url = new URL("jar:http://localhost:"
                           + new Integer(server.getAddress().getPort()).toString()
                           + "/deletetemp.jar!/");
-        JarURLConnection c = (JarURLConnection)url.openConnection();
-        JarFile f = c.getJarFile();
-        check(f.getEntry("entry") != null);
-        System.out.println(f.getName());
-        server.stop(0);
+            JarURLConnection c = (JarURLConnection)url.openConnection();
+            JarFile f = c.getJarFile();
+            check(f.getEntry("entry") != null);
+            System.out.println(f.getName());
+        } finally {
+            if (server != null) {
+                server.stop(0);
+            }
+        }
```

Comment:
Is there a reason to put the server.start in the try block, I would think this is simpler:

```
server.start();
try {
    :
} finally {
    server.stop(0);
}
```

Reply:
Thank you @jaikiran , @AlanBateman for your reviews. I have updated the patch with your inputs.